python/lib/dependabot/python/file_parser/pyproject_files_parser.rb

```diff
@@ -59,6 +67,38 @@ def pyproject_dependencies
           dependencies
         end
 
+        def pep621_dependencies
+          dependencies = Dependabot::FileParsers::Base::DependencySet.new
+
+          parsed_pep621_dependencies.each do |dep|
+            # If a requirement has a `<` or `<=` marker then updating it is
+            # probably blocked. Ignore it.
+            next if dep["markers"].include?("<")
```

Comment:
Not fully sure what to do with markers. If I understand correctly, this is what we currently do:

* For requirements.txt: https://github.com/dependabot/dependabot-core/blob/7b6f72eddeee3a4cf43ef61e13782b855010bcc0/python/lib/dependabot/python/file_parser.rb#L120-L126
* For setup.py: https://github.com/dependabot/dependabot-core/blob/7b6f72eddeee3a4cf43ef61e13782b855010bcc0/python/lib/dependabot/python/file_parser/setup_file_parser.rb#L30-L32
* For poetry I think we don't ignore them but maybe because we delegate to poetry itself which does the right thing.

I guess it makes sense to ignore marked dependencies in a similar way here too.

Reply:
> If a requirement has a `<`, `<=` or '==' marker then updating it is probably blocked. Ignore it.

I'm surprised we just ignore these dependencies for Python. For npm we check whether we have conflicting dependencies by passing them into the same library that the CLI uses https://github.com/dependabot/dependabot-core/blob/main/npm_and_yarn/helpers/lib/npm/conflicting-dependency-parser.js

Fixing this in Python is probably outside of the scope of this PR, but I wonder how many updates are blocked because we ignore certain dependency requirements

Reply:
Note that these are not requirements though, these are something Python calls ["environment markers"](https://peps.python.org/pep-0496/), which set additional constraints on a dependency in addition to version, like platform or python version. The idea (I believe) is that if there's a marker explicitly holding the dependency is that the user probably don't want an update because it won't work? Not sure.

Reply:
We'll be leaving this as is for now, marking as solved.